File: .github/scripts/add_type_to_existing.py
"""
Script para añadir type: aforismo a todos los archivos existentes del vault
que no tengan el campo type definido.
"""
import os
import glob
import sys
try:
    import frontmatter
except ImportError:
    print("Instalando python-frontmatter...")
    os.system("pip install python-frontmatter")
    import frontmatter

def main():
    vault_path = "content/vault"
    if not os.path.exists(vault_path):
        print("El directorio de notas no existe.")
        sys.exit(1)

    archivos = glob.glob(os.path.join(vault_path, "*.md"))
    updated = 0
    skipped = 0

    for archivo in archivos:
        basename = os.path.basename(archivo)
        # Ignorar archivos especiales
        if basename.startswith('_'):
            continue
        if basename.lower() in ['mi calendario.md', 'calendario.md']:
            continue

        try:
            with open(archivo, "r", encoding="utf-8") as f:
                post = frontmatter.load(f)

            if 'type' not in post.metadata:
                post.metadata['type'] = 'aforismo'
                with open(archivo, "w", encoding="utf-8") as f:
                    f.write(frontmatter.dumps(post))
                updated += 1
                print(f"  ✓ {basename}")
            else:
                skipped += 1
        except Exception as e:
            print(f"  ✗ Error en {basename}: {e}")

    print(f"\nResultado: {updated} archivos actualizados, {skipped} ya tenían type.")

if __name__ == "__main__":
    main()

File: .github/scripts/add_type_to_existing.js
#!/usr/bin/env node
/**
 * Añade type: aforismo a todos los archivos del vault que no tengan el campo type.
 * No requiere dependencias externas — usa solo el fs nativo de Node.js.
 */

const fs = require('fs');
const path = require('path');

const vaultPath = path.join(__dirname, '../../content/vault');

if (!fs.existsSync(vaultPath)) {
  console.error('❌ No se encontró content/vault');
  process.exit(1);
}

const SKIP = ['_mi calendario.md', 'calendario.md'];

const files = fs.readdirSync(vaultPath).filter(f => {
  if (!f.endsWith('.md')) return false;
  if (f.startsWith('_')) return false;
  if (SKIP.includes(f.toLowerCase())) return false;
  return true;
});

let updated = 0;
let skipped = 0;

for (const file of files) {
  const filePath = path.join(vaultPath, file);
  let content = fs.readFileSync(filePath, 'utf8');

  // Comprobar si ya tiene frontmatter
  const hasFrontmatter = content.startsWith('---');
  
  if (!hasFrontmatter) {
    // Sin frontmatter: añadir uno nuevo con type
    content = `---\ntype: aforismo\n---\n\n${content.trimStart()}`;
    fs.writeFileSync(filePath, content, 'utf8');
    console.log(`  ✓ [nuevo FM] ${file}`);
    updated++;
    continue;
  }

  // Con frontmatter: comprobar si ya tiene type
  const fmEnd = content.indexOf('\n---', 3);
  if (fmEnd === -1) { skipped++; continue; }

  const fmBlock = content.slice(0, fmEnd + 4);
  
  if (/^type:/m.test(fmBlock)) {
    skipped++;
    continue; // Ya tiene type, no tocar
  }

  // Insertar type justo después de la primera línea ---
  const firstLineEnd = content.indexOf('\n');
  const before = content.slice(0, firstLineEnd + 1);
  const after = content.slice(firstLineEnd + 1);
  content = `${before}type: aforismo\n${after}`;
  
  fs.writeFileSync(filePath, content, 'utf8');
  console.log(`  ✓ ${file}`);
  updated++;
}

console.log(`\n✅ Resultado: ${updated} archivos actualizados, ${skipped} ya tenían type.`);
